### PR TITLE
Live configuration reload will change logging level

### PR DESCRIPTION
**Pull request by itaiad200**
lakeFS reads the logging level when it is changed in the configuration file, and change the logging on the fly without restarting the app.

To enable more configuration properties to change, we should propagate them properly in the same manner. Since this isn't always trivial, I suggest we do it per property, when needed.

### CODE REVIEW

**Review by guy-har — APPROVED** [review]
Cool!

File: cmd/lakefs/cmd/run.go [hunk 1]
```diff
@@ -91,6 +93,13 @@ var runCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		logger := logging.Default()
 		cfg := loadConfig()
+		viper.WatchConfig()
+		viper.OnConfigChange(func(in fsnotify.Event) {
+			lvl := viper.GetString("logging.level")
+			fmt.Println("Changing log level to", lvl)
+			logging.SetLevel(lvl)
```
Comment (Jonathan-Rosenberg) [inline cmd/lakefs/cmd/run.go:100]: Do we want to handle an empty `lvl` string (in case that `GetString` returned such one)?

Comment (itaiad200) [reply cmd/lakefs/cmd/run.go:100]: We handle it here the same way we handle it on startup -> Let viper decide.

File: cmd/lakefs/cmd/run.go [hunk 2]
```diff
@@ -91,6 +93,13 @@ var runCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		logger := logging.Default()
 		cfg := loadConfig()
+		viper.WatchConfig()
+		viper.OnConfigChange(func(in fsnotify.Event) {
+			lvl := viper.GetString(config.LoggingLevelKey)
```
Comment (arielshaqed) [inline cmd/lakefs/cmd/run.go:98]: Can we (additionally) read the rest of the config, and warn if something else changed?

File: cmd/lakefs/cmd/run.go (hunk 1 above)
Comment (arielshaqed) [reply cmd/lakefs/cmd/run.go:100]: What do we do if SetLevel failed?  I think we should at least issue an error here.

File: cmd/lakefs/cmd/run.go [hunk 3]
```diff
@@ -91,6 +93,13 @@ var runCmd = &cobra.Command{
...
+			fmt.Println("Changing log level to", lvl)
```
Comment (arielshaqed) [inline cmd/lakefs/cmd/run.go:99]: This should be a log, with times and everything.

**Review by arielshaqed — APPROVED** [review]
Cool!

File: cmd/lakefs/cmd/run.go (hunk 1 above)
Comment (itaiad200) [reply cmd/lakefs/cmd/run.go:100]: It can't fail and it doesn't return an error.

File: cmd/lakefs/cmd/run.go (hunk 2 above)
Comment (itaiad200) [reply cmd/lakefs/cmd/run.go:98]: Opening an issue, there's a lot to do here (save the previous config, compare, take defaults into account while only the logging level can really change), while the value is limited IMO (Better visibility for items that don't have any effect).

**Pull request merged**